arelle/ArelleBaseOptions.py

```diff
@@ -0,0 +1,361 @@
+                 keepOpen: Optional[bool] = None,
+                 labelLang: Optional[str] = None,
+                 labelRole: Optional[str] = None,
+                 logCodeFilter: Optional[str] = None,
+                 logFile: Optional[str] = None,
+                 logFormat: Optional[str] = None,
+                 logLevel: Optional[str] = None,
+                 logLevelFilter: Optional[str] = None,
+                 logRefObjectProperties: Optional[str] = None,
+                 logTextMaxLength: Optional[int] = None,
+                 monitorParentProcess: Optional[bool] = None,
+                 noCertificateCheck: Optional[bool] = None,
+                 outputAttribution: Optional[str] = None,
+                 packageManifestName: Optional[str] = None,
+                 packages: Optional[str] = None,
+                 parameterSeparator: Optional[str] = None,
+                 parameters: Optional[str] = None,
+                 password: Optional[str] = None,
+                 plugins: Optional[str] = None,
+                 preFile: Optional[str] = None,
+                 proxy: Optional[str] = None,
+                 relationshipCols: Optional[int] = None,
+                 roleTypesFile: Optional[str] = None,
+                 rssReport: Optional[str] = None,
+                 rssReportCols: Optional[int] = None,
+                 saveTargetFiling: Optional[bool] = None,
+                 saveTargetInstance: Optional[bool] = None,
+                 showEnvironment: Optional[bool] = None,
+                 showOptions: Optional[bool] = None,
+                 skipDTS: Optional[bool] = None,
+                 skipExpectedInstanceComparison: Optional[bool] = None,
+                 skipLoading: Optional[bool] = None,
+                 statusPipe: Optional[str] = None,
+                 tableFile: Optional[str] = None,
+                 testReport: Optional[str] = None,
+                 testReportCols: Optional[int] = None,
+                 testcaseResultOptions: Optional[str] = None,
+                 testcaseResultsCaptureWarnings: Optional[bool] = None,
+                 timeVariableSetEvaluation: Optional[bool] = None,
+                 uiLang: Optional[str] = None,
+                 username: Optional[str] = None,
+                 utrUrl: Optional[str] = None,
+                 utrValidate: Optional[bool] = None,
+                 validate: Optional[bool] = None,
+                 validateEFM: Optional[bool] = None,
+                 validateEFMCalcTree: Optional[bool] = None,
+                 validateHMRC: Optional[bool] = None,
+                 validateTestcaseSchema: Optional[bool] = None,
+                 versReportFile: Optional[str] = None,
+                 viewArcrole: Optional[bool] = None,
+                 viewFile: Optional[str] = None,
+                 webserver: Optional[str] = None,
+                 xdgConfigHome: Optional[str] = None,
+                 **kwargs):
+        self.DTSFile = DTSFile
```

Comment:
There's a lot of argument duplication. You should be able to define these once using a dataclass and [post init processing](https://docs.python.org/3/library/dataclasses.html#post-init-processing) to handle plugins and validation

Reply:
Something like this:
```python
from __future__ import annotations

from dataclasses import InitVar, dataclass
from typing import Optional, Union

from arelle.typing import TypeGetText

_: TypeGetText

RuntimeOptionValue = Union[bool, int, str, None]


class RuntimeOptionsException(Exception):
    pass


@dataclass()
class RuntimeOptions:
    pluginOptions: InitVar[Optional[dict[str, RuntimeOptionValue]]] = None

    abortOnMajorError: Optional[bool] = None
    about: Optional[str] = None
    # ...

    def __post_init__(self, pluginOptions: Optional[dict[str, RuntimeOptionValue]]) -> None:
        if pluginOptions:
            existingBaseOptions = sorted({
                optionName
                for optionName in pluginOptions.keys()
                if hasattr(self, optionName)
            })
            if existingBaseOptions:
                raise RuntimeOptionsException(_('Provided plugin options already exist as base options {}'.format(existingBaseOptions)))
            for optionName, optionValue in pluginOptions.items():
                setattr(self, optionName, optionValue)

        if self.entrypointFile is None and ...:
            raise RuntimeOptionsException(_('Incorrect arguments'))

        if self.webserver and ...:
            raise RuntimeOptionsException(_('Incorrect arguments with webserver'))

```